serde/src/main/java/com/amazon/ionhiveserde/objectinspectors/IonValueToStringObjectInspector.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.ionhiveserde.objectinspectors;
+
+import com.amazon.ion.IonText;
+import com.amazon.ion.IonType;
+import com.amazon.ion.IonValue;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.StringObjectInspector;
+import org.apache.hadoop.hive.serde2.typeinfo.TypeInfoFactory;
+import org.apache.hadoop.io.Text;
+
+/**
+ * Adapts an {@link IonText} for the string Hive type.
+ */
+public class IonValueToStringObjectInspector extends AbstractIonPrimitiveJavaObjectInspector implements
+        StringObjectInspector {
+
+    public IonValueToStringObjectInspector() {
+        super(TypeInfoFactory.stringTypeInfo);
+    }
+
+    @Override
+    public String getPrimitiveJavaObject(final Object o) {
+        if (IonUtil.isIonNull((IonValue) o)) {
+            return null;
+        }
+
+        return getPrimitiveWritableObject((IonValue) o);
+    }
+
+    @Override
+    public Text getPrimitiveWritableObject(final Object o) {
+        if (IonUtil.isIonNull((IonValue) o)) {
+            return null;
+        }
+
+        return new Text(getPrimitiveWritableObject((IonValue) o));
+    }
+
+    private String getPrimitiveWritableObject(final IonValue ionValue) {
+        String s = ionValue.toString();
+        return ionValue.getType() == IonType.STRING ? s.substring(1, s.length() - 1) : s;
```

Comment:
Using substring to extract desired Ion string since it returns a string with Double quotes. (e.g. returns `\"string\"` instead of `string`).

Reply:
After conversation with @tgregg it looks like the right idiom to use here might be something like:

```java
if (IonType.isText(value.getType()) {
  return value.stringValue();
}
return value.toString();
```

That will get us String content `foo` for the Ion string represented `"foo"` or the Ion symbol representable as `'foo'` or `foo`, but get us e.g. String content `{ foo: "bar" }` for an Ion struct. I think this is the best we can do for principle of least surprise.

Reply:
changed.